### PR TITLE
Onboard clang into build scripts and CI

### LINKED ISSUE #696: Onboard a Linux Clang build to CI
Clang is becoming more and more popular as a compiler on linux, and its a bit more particular then GCC. We should onboard a CI build using clang to ensure the build doesn't break there for external users.

### PR DESCRIPTION
**Pull request by ThadHouse**
Closes #696

### CODE REVIEW

File: .azure/templates/build-config-user.yml [hunk 1]
```diff
@@ -8,6 +8,7 @@ parameters:
   tls: ''
   config: 'Debug,Release'
   extraBuildArgs: ''
+  skipArtifacts: ''
```
Comment (nibanks) [inline .azure/templates/build-config-user.yml:11]: Default to `false` instead?

File: .azure/azure-pipelines.ci.yml [hunk 2]
```diff
@@ -134,6 +134,20 @@ stages:
       tls: openssl
       extraBuildArgs: -DisableLogs -Generator Ninja -ToolchainFile cmake/toolchains/arm-pi-gnueabihf.toolchain.cmake
 
+# Unsupported currently.
+- stage: build_linux_clang
```
Comment (nibanks) [inline .azure/azure-pipelines.ci.yml:138]: I'd prefer to not make a separate stage...

Comment (ThadHouse) [reply .azure/azure-pipelines.ci.yml:138]: Then I have to modify the build-config-user.yml to have an extra string I can use to modify the job name. I can do that if you think that'd be cleaner.

Comment (nibanks) [reply .azure/azure-pipelines.ci.yml:138]: Yeah, I think I prefer that way.

**Pull request merged**